Tests/CLICommands/RunCommandHandlerSpec.swift

```diff
@@ -0,0 +1,112 @@
+                                                          state: expectedState)
+                    
+                    runCommandResult = runCommandHandler.handle()
+                }
+                
+                it("runs all its steps") {
+                    expect(stepSpy1.methodCalls) == ["run(with:)"]
+                    expect(stepSpy2.methodCalls) == ["run(with:)"]
+                    expect(stepSpy3.methodCalls) == ["run(with:)"]
+                }
+                
+                it("shares the same state across all of the steps") {
+                    expect(stepSpy1.states.first) === expectedState
+                    expect(stepSpy2.states.first) === expectedState
+                    expect(stepSpy3.states.first) === expectedState
+                }
+                
+                it("applies any state mutations returned by the steps") {
+                    expect(expectedState.tempDirectoryURL) == URL(string: "/lol/son")!
+                    expect(expectedState.projectDirectoryURL) == URL(string: "/lol")!
+                }
+                
+                it("returns success") {
+                    guard case .success(_) = runCommandResult! else {
+                        fail("expected success but got \(String(describing: runCommandResult!))")
+                        return
+                    }
+                }
+                
+            }
+            
+            context("when there is a failure in one of its steps") {
+                beforeEach {
+                    stepSpy1 = RunCommandStepSpy()
+                    stepSpy1.resultToReturn = .success([])
+                    
+                    stepSpy2 = RunCommandStepSpy()
+                    stepSpy2.resultToReturn = .failure(.configurationParsingError(reason: ""))
+                    
+                    stepSpy3 = RunCommandStepSpy()
+                    stepSpy3.resultToReturn = .success([])
+                    
+                    expectedState = RunCommandState()
+                    
+                    runCommandHandler = RunCommandHandler(steps: [stepSpy1,
+                                                                  stepSpy2,
+                                                                  stepSpy3],
+                                                          state: expectedState)
+                    
+                    runCommandResult = runCommandHandler.handle()
+                }
+                
+                it("won't run any subsequent steps after the failing step") {
+                    expect(stepSpy1.methodCalls) == ["run(with:)"]
+                    expect(stepSpy2.methodCalls) == ["run(with:)"]
+                    expect(stepSpy3.methodCalls).to(beEmpty())
+                }
+                
+                it("won't share the state with any subsequent steps after the failing step") {
+                    expect(stepSpy1.states.first) === expectedState
```

Comment:
I think we could use `beIdenticalTo()`?

` expect(stepSpy1.states.first).to(beIdenticalTo(expectedState))`